packages/parameters/tests/unit/setParameter.test.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Test setParameter function
+ *
+ * @group unit/parameters/ssm/setParameter/function
+ */
+import { PutParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+import { mockClient } from 'aws-sdk-client-mock';
+import { DEFAULT_PROVIDERS } from '../../src/base';
+import { SSMProvider } from '../../src/ssm/SSMProvider';
+import { setParameter } from '../../src/ssm/setParameter';
+import 'aws-sdk-client-mock-jest';
+import type { SSMSetOptions } from '../../src/types/SSMProvider';
+
+describe('Function: setParameter', () => {
+  const parameterName = '/my-parameter';
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('when called and a default provider does not exist, it instantiates one and sets the parameter', async () => {
+    const options: SSMSetOptions = { value: 'my-value' };
+    const client = mockClient(SSMClient)
```

Comment:
In other tests (i.e. `getParameter.test.ts`) we do it like here, but we can improve these tests by creating the client once around L16 and then reusing it across test cases. To do this, we'll need to add an `afterEach` section like described [here](https://docs.powertools.aws.dev/lambda/typescript/latest/utilities/parameters/#handlertestts_2) so we can reset the client between tests.

This should improve performance and reduce some boilerplate.